### PR TITLE
FEATURE: Workspace aware content cache

### LINKED ISSUE #4810: BUG: ContentCache entries and keys are not workspace/contentstream aware
In Neos <9.0 we stored the workspace name of a node, within the content cache key to be able to flush caches more precisely and prevent exposing cached user-workspace content to live.

We need to discuss, what fits best here: **workspace name** or **contentstream id**
And add the addional identifier to the cache keys in cache creation and flushing.

We also should check if it's still needed to iterate over workspace hirarchy for cache flushing, as we do it in Neos <9.0.

### PR DESCRIPTION
**Pull request by dlubitz**
Changes the cache tags from contentStreamIds to workspaceNames.

Fixes #4810

### CODE REVIEW

File: Neos.Neos/Classes/Fusion/Cache/ContentCacheFlusher.php [hunk 1]
```diff
@@ -125,7 +115,7 @@ private function collectTagsForChangeOnNodeAggregate(
             $parentNodeAggregates[] = $parentNodeAggregate;
         }
         // we do not need these variables anymore here
-        unset($contentStreamId, $nodeAggregateId);
+        unset($nodeAggregateId);
```
Comment (mhsdesign) [inline Neos.Neos/Classes/Fusion/Cache/ContentCacheFlusher.php:118]: lol

File: Neos.Neos/Tests/Behavior/Features/ContentCache/Nodes.feature [hunk 2]
```diff
@@ -179,6 +181,41 @@ Feature: Tests for the ContentCacheFlusher and cache flushing on node and nodety
     cacheVerifier=first execution, title=Node a1
     """
 
+  Scenario: ContentCache gets flushed when a property of another node has changed that is added as nodeTagForIdentifier
```
Comment (mhsdesign) [inline Neos.Neos/Tests/Behavior/Features/ContentCache/Nodes.feature:185]: this tests doesnt make sense to me when reading ^^

**Review by mhsdesign — COMMENTED** [review]
Thanks for taking care ;) Left some comments. I think we can also just do the todo followup to make the asset usage workspace aware in a new pr as this works standalone :)

File: Neos.Neos/Tests/Behavior/Features/ContentCache/Nodes.feature (hunk 2 above)
Comment (dlubitz) [reply Neos.Neos/Tests/Behavior/Features/ContentCache/Nodes.feature:185]: It's about testing this line:
https://github.com/neos/neos-development-collection/pull/5055/files#diff-8502ed8e7ef0f4168128ab64972a41831434a3fba4650271672a75e29506db11R86

Comment (dlubitz) [reply Neos.Neos/Tests/Behavior/Features/ContentCache/Nodes.feature:185]: The cache needs to get flushed on every change that happens to a3

Comment (mhsdesign) [reply Neos.Neos/Tests/Behavior/Features/ContentCache/Nodes.feature:185]: ah lol yes now it makes sense is just a bit confusing but thanks for the test :D (you could add a commend like `see {Neos.Caching.nodeTagForIdentifier('a3',node) in setup and now well update it`

**Review by mhsdesign — APPROVED** [review]

**Comment by mhsdesign** [comment]
Moved the todo to https://github.com/neos/neos-development-collection/issues/5043 and we can merge oke?

**Pull request merged**